src/App.js

```diff
@@ -43,18 +45,22 @@ function App() {
     console.log(cartProducts)
   };
 
-  const onAddToFavorite = (obj) => {
-    setFavorites(prev => {
-      // Check if the item already exists in favorites
-      if (!prev.some(item => item.id === obj.id)) {
-        // If it doesn't exist, add it to the favorites array
-        return [...prev, obj];
-      }
-      // If it already exists, return the unchanged favorites array
-      return prev;
-    });
 
-  };
+  const onAddToFavorite = async (obj) => {
+    if (obj.name) {
+      try {
+        if ((favorites).find(item => Number(item.id) === Number(obj.id))) {
+          axios.delete(`https://64707a693de51400f7245522.mockapi.io/favorites/${obj.id}`);
```

Comment:
let's move https://64707a693de51400f7245522.mockapi.io to const instead of copying

Reply:
done